frontend/packages/ux-editor/src/components/config/editModal/EditOptions/OptionTabs/EditTab/EditTab.tsx

```diff
@@ -70,43 +69,14 @@ function SelectedOptionList({
   component,
   handleComponentChange,
 }: SelectedOptionListProps) {
-  const { t } = useTranslation();
-
-  const handleDelete = () => {
-    if (component.options) {
-      delete component.options;
-    }
-
-    const emptyOptionsId = '';
-    handleComponentChange({
-      ...component,
-      optionsId: emptyOptionsId,
-    });
-
-    setComponentHasOptionList(false);
-  };
-
-  const label =
-    component.optionsId !== '' && component.optionsId !== undefined
-      ? component.optionsId
-      : t('ux_editor.modal_properties_code_list_custom_list');
-
   return (
-    <div aria-label={label} className={classes.chosenOptionContainer}>
+    <div className={classes.chosenOptionContainer}>
       <OptionListEditor
-        label={label}
         optionsId={component.optionsId}
         component={component}
         handleComponentChange={handleComponentChange}
+        setComponentHasOptionList={setComponentHasOptionList}
```

Comment:
If this value is defined by the incoming components fields, should we not be able to evaluate this in a const used only where it is needed? And any changes to the component that would affect this value would anyway be noticed due to rerendring affected by changing the component-prop 🤔

Reply:
I added the useState to make sure we continue to render the modals while the user is making changes. Removing it results in the modals being closed because they are no longer being rendered as they get replaced with `SelectedOptionList`

If you have a better idea for how to fix this, then i'm happy to change it